hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieAvroHFileReader.java

```diff
@@ -120,20 +120,14 @@ public HoodieAvroHFileReader(Path path, HFile.Reader reader, Option<Schema> sche
         .orElseGet(() -> Lazy.lazily(() -> fetchSchema(reader)));
   }
 
-  @Override
-  public Option<HoodieRecord<IndexedRecord>> getRecordByKey(String key, Schema readerSchema) throws IOException {
-    synchronized (sharedScannerLock) {
-      return fetchRecordByKeyInternal(sharedScanner, key, getSchema(), readerSchema)
-          .map(data -> unsafeCast(new HoodieAvroIndexedRecord(data)));
-    }
-  }
 
   @Override
   public ClosableIterator<HoodieRecord<IndexedRecord>> getRecordsByKeysIterator(List<String> keys, Schema schema) throws IOException {
     // We're caching blocks for this scanner to minimize amount of traffic
     // to the underlying storage as we fetched (potentially) sparsely distributed
     // keys
     HFileScanner scanner = getHFileScanner(reader, true);
+    scanner.seekTo(); // places the cursor at the beginning of the first data block.
     ClosableIterator<IndexedRecord> iterator = new RecordByKeyIterator(scanner, keys, getSchema(), schema);
```

Comment:
Should we always invoke `scanner.seekTo` just inside `getHFileScanner`, right after we got a reader.

Reply:
yes, makes sense.

Reply:
For full scan, we should always seek to the start again? I just though it is hard to maintain the `seekTo` for each caller.

Reply:
sounds good. will fix it.